app/grandchallenge/evaluation/models.py

```diff
@@ -1715,6 +1730,42 @@ def get_absolute_url(self):
             },
         )
 
+    @cached_property
+    def has_matching_algorithm_interfaces(self):
+        return sorted(
+            [int.pk for int in self.phase.algorithm_interfaces.all()]
+        ) == sorted(
+            [int.pk for int in self.algorithm_image.algorithm.interfaces.all()]
+        )
```

Comment:
Easier as a set comparison?

```suggestion
        return (
            {*self.phase.algorithm_interfaces.all()} == {*self.algorithm_image.algorithm.interfaces.all}
        )
```